app/web/components/LanguagePickerSelect.tsx

```diff
@@ -0,0 +1,195 @@
+  const theme = useTheme();
+  const isMdOrWider = useMediaQuery(theme.breakpoints.up("md"));
+
+  const handleChange = (event: SelectChangeEvent) => {
+    const newLang = event.target.value as string;
+    setLanguage(newLang);
+    onSelect?.(newLang); // sends request to update language preference on backend?
+  };
+
+  // Helper function to render a flag icon from country flag icons collection
+  const renderFlag = (flagCode: string) => {
+    return (
+      <img
+        alt={`${flagCode} flag`}
+        src={`http://purecatamphetamine.github.io/country-flag-icons/3x2/${flagCode}.svg`}
+        style={{ width: 30 }}
+      />
+    );
+  };
+
+  // Build list of menu items based on LANGUAGE_MAP
+  const menuItems: React.ReactNode[] = [];
+  for (const languageCode in LANGUAGE_MAP) {
+    const flagCode = LANGUAGE_MAP[languageCode].flagIconCode;
+
+    menuItems.push(
+      <MenuItem
+        value={languageCode}
+        sx={{
+          display: "flex",
+          alignItems: "center",
+          gap: theme.spacing(1),
+          "& .Mui-selected": {
+            backgroundColor: theme.palette.action.selected,
+          },
+          "& .Mui-selected:hover": {
+            backgroundColor: theme.palette.action.hover,
+          },
+        }}
+      >
+        <Stack
+          sx={{ width: "100%" }}
+          direction="row"
+          alignItems="center"
+          justifyContent="space-between"
+        >
+          <Stack direction="row">
+            <ListItemIcon>{renderFlag(flagCode)}</ListItemIcon>
+            <ListItemText
+              sx={{ color: "#666666", fontWeight: "bold", display: "inline" }}
+            >
+              {languageCode.toUpperCase()}
+            </ListItemText>
+          </Stack>
+          {/* if this menu item matches selected language, display a check mark */}
+          <div>
+            {language === languageCode && (
+              <CheckIcon fontSize="small" sx={{ color: "#00a69a" }} />
+            )}
+          </div>
+        </Stack>
+      </MenuItem>,
+    );
+  }
+
+  return (
+    <Box sx={{ minWidth: 60 }}>
+      <FormControl
+        variant="outlined"
+        displayMode={displayMode}
+        sx={{
+          // specialized sizing based on screen size
+          width:
+            displayMode === "round"
+              ? "fit-content"
+              : isMdOrWider
+                ? "241px"
+                : "100%",
+        }}
+      >
+        {" "}
+        {displayMode === "round" ? (
+          <MuiSelect
+            id="language-select"
+            value={language}
+            sx={{
+              borderRadius:
+                displayMode === "round" ? 999 : theme.shape.borderRadius,
+              "& .MuiSelect-icon": {
+                color: theme.palette.text.primary,
+                fontSize: "1.25rem",
+                top: "50%",
+                transform: "translateY(-50%)",
+                right: 10,
+              },
+            }}
```

Comment:
This one might be worth throwing in a StyledSelect and moving the css there as there's a bit more.